### DISCUSSION TITLE
removeLiquidityETHSupportingFeeOnTransferToken

### DISCUSSION THREAD

**Discussion by Aruninnblockchain** [Uniswap/interface#2244] [General]
Why we need removeLiquidityETHSupportingFeeOnTransferToken function?

How it works? I couldn't found an exact answer in many forums. As a Developer when should i call removeLiquidtyETH( ) and removeLiquidityETHSupportingFeeOnTransferToken( ) function?  If you provide exact answer, this would be helpful for many people. Most of the people are searching answer for the above one.

**Comment by Kingtsu3288** [comment]
> Why we need removeLiquidityETHSupportingFeeOnTransferToken function?
> 
> 
> 
> How it works? I couldn't found an exact answer in many forums. As a Developer when should i call removeLiquidtyETH( ) and removeLiquidityETHSupportingFeeOnTransferToken( ) function?  If you provide exact answer, this would be helpful for many people. Most of the people are searching answer for the above one.

- [ ] @jab416171 ![image](https://user-images.githubusercontent.com/82915595/135818869-3120aaed-10ad-4fe2-bd24-9bc68c06c498.jpeg)

**Reply by VeronikaSemykopna** [reply]
🦁
![image](https://github.com/Uniswap/interface/assets/137341607/6d7848be-0465-49da-811a-757ebfd2706f)

**Comment by sambacha** [comment]
> Why we need removeLiquidityETHSupportingFeeOnTransferToken function?
> 
> How it works? I couldn't found an exact answer in many forums. As a Developer when should i call removeLiquidtyETH( ) and removeLiquidityETHSupportingFeeOnTransferToken( ) function? If you provide exact answer, this would be helpful for many people. Most of the people are searching answer for the above one.

## There is an entire section in the audit reports discussing this [https://uniswap.org/audit.html#orgd15839d](https://uniswap.org/audit.html#orgd15839d)

> Router: incompatible with token with fees on transfer
>The user facing contract UniswapV2Router01.sol calculates the amount a user needs to transfer to UniswapV2Pair.sol in order to perform a mint, swap or burn.

> In cases dealing with a token which subtracts a fee on transferFrom, the amount received by UniswapV2Pair is smaller than the required amount to perform a successful swap, causing the whole call to fail when the fee adjusted invariant is checked at the end of swap.

> This is only a problem with the router, and additional routers or token wrappers can be used to mitigate this issue.
